app/src/main/java/edu/grinnell/appdev/grinnelldirectory/Activity/LoginActivity.java

```diff
@@ -35,9 +46,61 @@ protected void onCreate(Bundle savedInstanceState) {
      *
      * @param view LoginActivity's view
      */
-    @OnClick(R.id.signin)
-    public void signIn(View view) {
-        String usernameTxt = mUsernameEditText.getText().toString();
-        String passwordTxt = mPasswordEditText.getText().toString();
+    @OnClick(R.id.login)
+    void signIn(View view) {
+        String username = mUsernameEditText.getText().toString();
+        String password = mPasswordEditText.getText().toString();
+        User user = new User(username, password);
+        DBAPICaller dbapiCaller = new DBAPICaller(user, this);
+        dbapiCaller.authenticateUser();
+    }
+
+    @Override public void onSearchSuccess(List<Person> people) {}
+
+    /**
+     * Save credentials and move to MainActivity if login succeeded
+     *
+     * @param success whether the login was successful
+     * @param person model for the logged in user
+     */
+    @BindString(R.string.authentication_failure) String authenticationFailure;
+    @Override public void authenticateUserCallSuccess(boolean success, Person person) {
+        final String username = mUsernameEditText.getText().toString();
+        final String password = mPasswordEditText.getText().toString();
```

Comment:
@as-f So what if the user changes the username or password field between when the button is clicked and when the response is received? Think about what will actually be saved to the system. What is the best way to handle it?